### PR TITLE
Splinter keygen group option

### PR DESCRIPTION
**Pull request by Caleb-Hill**
This adds the `--group` flag to the splinter keygen command.

### CODE REVIEW

File: cli/src/action/keygen.rs [hunk 1]
```diff
@@ -65,6 +65,18 @@ impl FromStr for GroupOptions {
 impl Action for KeyGenAction {
     fn run<'a>(&mut self, arg_matches: Option<&ArgMatches<'a>>) -> Result<(), CliError> {
         let args = arg_matches.ok_or(CliError::RequiresArgs)?;
+        let group: Option<GroupOptions> = args
+            .value_of("group")
+            .map(|s| Some(GroupOptions::from_str(s)))
```
[the remark this answers is not in the document]
Comment (Caleb-Hill) [reply cli/src/action/keygen.rs]: No reason, fixed

File: cli/src/action/keygen.rs [hunk 2]
```diff
@@ -38,6 +39,29 @@ const DEFAULT_SYSTEM_KEY_NAME: &str = "splinterd";
...
+    Auto,
+    Named(String),
+    GroupID(u32),
+}
+
+impl FromStr for GroupOptions {
+    type Err = CliError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "auto" => Ok(GroupOptions::Auto),
```
Comment (peterschwarz) [inline cli/src/action/keygen.rs:59]: If this is only for parsing the clap args, you should just parse this value there.  See the recent change with ScabbardState

**Review by peterschwarz — CHANGES_REQUESTED** [review]

File: cli/src/action/keygen.rs (hunk 2 above)
Comment (Caleb-Hill) [reply cli/src/action/keygen.rs]: Fixed

File: cli/src/action/keygen.rs [hunk 3]
```diff
@@ -148,7 +149,7 @@ fn write_keys(
     public_key_path: PathBuf,
     force_create: bool,
     skip_create: bool,
-    change_permissions: bool,
+    group: Option<GroupOptions>,
```
Comment (peterschwarz) [inline cli/src/action/keygen.rs]: As this previously was always passing true, should this still be an option?  If no option is provided on the command line, the behaviour of this command will change. Given that `Auto` is the previous behaviour, perhaps this should not be optional and the arg should be or-elsed to that.

Comment (peterschwarz) [reply cli/src/action/keygen.rs]: After some discussion, this bahaviour change is intentional.

**Review by peterschwarz — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Pull request merged**